CMakeLists.txt

```diff
@@ -248,6 +248,8 @@ if( BUILD_PACKAGE )
 endif()
 
 if (BUILD_DOCS)
+  option(BUILD_DOCS_ONLY_WITH_PACKAGE "Build DOCS only explicitly via `make docs` or when `make package`" OFF)
+  # mark_as_advanced(BUILD_DOCS_ONLY_WITH_PACKAGE)
```

Comment:
New option. Wasn't sure if it should be made 'advanced' or not, I choose not to (so people know it's a thing)